kubernetes/resource_kubernetes_service_account.go

```diff
@@ -0,0 +1,220 @@
+package kubernetes
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgApi "k8s.io/apimachinery/pkg/types"
+	api "k8s.io/kubernetes/pkg/api/v1"
+	kubernetes "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+func resourceKubernetesServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKubernetesServiceAccountCreate,
+		Read:   resourceKubernetesServiceAccountRead,
+		Exists: resourceKubernetesServiceAccountExists,
+		Update: resourceKubernetesServiceAccountUpdate,
+		Delete: resourceKubernetesServiceAccountDelete,
+
+		// This resource is not importable because the API doesn't offer
+		// any way to differentiate between default & user-defined secret
+		// after the account was created.
+
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("service account", true),
+			"image_pull_secret": {
+				Type:        schema.TypeSet,
+				Description: "A list of references to secrets in the same namespace to use for pulling any images in pods that reference this Service Account. More info: http://kubernetes.io/docs/user-guide/secrets#manually-specifying-an-imagepullsecret",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "Name of the referent. More info: http://kubernetes.io/docs/user-guide/identifiers#names",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"secret": {
+				Type:        schema.TypeSet,
+				Description: "A list of secrets allowed to be used by pods running using this Service Account. More info: http://kubernetes.io/docs/user-guide/secrets",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "Name of the referent. More info: http://kubernetes.io/docs/user-guide/identifiers#names",
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"default_secret_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceKubernetesServiceAccountCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	svcAcc := api.ServiceAccount{
+		AutomountServiceAccountToken: ptrToBool(false),
+		ObjectMeta:                   metadata,
+		ImagePullSecrets:             expandLocalObjectReferenceArray(d.Get("image_pull_secret").(*schema.Set).List()),
+		Secrets:                      expandServiceAccountSecrets(d.Get("secret").(*schema.Set).List(), ""),
```

Comment:
We have no default name at this point yet (during initial creation), so there's nothing to add.